### PR TITLE
 ♻️  [RUM-67] Specialise type files

### PR DESCRIPTION
**Pull request by RomanGaignault**
## Motivation

Specialise types files to have a more descriptive name.

## Changes

Changed types name and import from different part of the SDK.

## Testing

- [ ] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/core/src/index.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export {
   deleteCookie,
   resetInitCookies,
 } from './browser/cookie'
-export { CookieStore, WeakRef, WeakRefConstructor } from './browser/types'
+export { CookieStore, WeakRef, WeakRefConstructor } from './browser/browser.types'
```
Comment (cy-moi) [inline packages/core/src/index.ts:96]: 💬 Suggestion: this looks a bit redundant and less helpful. Maybe we could go with the official TS file naming, such as, `dom.custom.types` or other more specific ones?

File: developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.types.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Those types are coming from the Web-UI Session Replay Player. Please keep them as close as
 // possible to the original types.
 
-import type { BrowserRecord, RecordType } from '../../../../packages/rum/src/types'
+import type { BrowserRecord, RecordType } from '@datadog/browser-rum/src/types'
```
Comment (cy-moi) [inline developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.types.ts:4]: 👏 Praise: good catch!

File: developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.ts [hunk 3]
```diff
@@ -2,8 +2,8 @@ import type { BrowserRecord } from '../../../../packages/rum/src/types'
 import { IncrementalSource, RecordType } from '../../../../packages/rum/src/types'
```
Comment (cy-moi) [inline developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.ts:2]: 💬 Suggestion: shall this also be `@datadog/browser-rum/src/types`?

**Review by cy-moi — APPROVED** [review]

File: developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.types.ts
Comment (amortemousque) [inline developer-extension/src/panel/sessionReplayPlayer/startSessionReplayPlayer.types.ts:1]: **💬 suggestion:** `sessionReplayPlayer.type.ts` should be enough :)‏

**Review by amortemousque — APPROVED** [review]

File: packages/core/src/browser/dom.custom.types.ts
Comment (amortemousque) [inline packages/core/src/browser/dom.custom.types.ts:1]: **💬 suggestion:** `browser.types.ts`‏

**Review by amortemousque — APPROVED** [review]

**Comment by amortemousque** [comment]
Nice!

File: packages/core/src/browser/dom.custom.types.ts
Comment (cy-moi) [reply packages/core/src/browser/dom.custom.types.ts:1]: 🙇‍♀️ Sorry, it was like this. I suggested the other way round.

**Pull request merged**